src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+	
+	@Autowired
+	private PermissibleValueService pvSvc;
+	
+	@Autowired
+	private DaoFactory daoFactory;
+	
+	private Map<String, CollectionProtocolDetail> cpMap = new HashMap<>();
+	
+	private Map<String, CollectionProtocolEventDetail> eventsMap = new HashMap<>();
+	
+	private Map<String, SpecimenRequirementDetail> srMap = new HashMap<>();
+	
+	@Override
+	public ResponseEvent<CarsDetail> importObject(RequestEvent<ImportObjectDetail<CarsDetail>> req) {
+		try {
+			ImportObjectDetail<CarsDetail> detail = req.getPayload();
+			importRecord(detail);
+			return ResponseEvent.response(detail.getObject());
+		} catch (OpenSpecimenException ose) {
+			return ResponseEvent.error(ose);
+		} catch (Exception e) {
+			return ResponseEvent.serverError(e);
+		}
+	}
+	
+	@PlusTransactional
+	private ResponseEvent<Object> importRecord(ImportObjectDetail<CarsDetail> detail) throws Exception {
+		CollectionProtocolEventDetail event = toEvent(detail.getObject());
+		
+		CollectionProtocolDetail cpDetail = checkCp(event, detail.getObject());
+		checkEvent(cpDetail, event);
+		checkSr(event);
+		
+		return null;
+	}
+
+	private CollectionProtocolEventDetail toEvent(CarsDetail detail) {
+		CollectionProtocolEventDetail eventDetail = new CollectionProtocolEventDetail();
+		SpecimenRequirementDetail srDetail = new SpecimenRequirementDetail();
+		
+		eventDetail.setCpShortTitle(detail.getIrbNumber());
+		eventDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		
+		srDetail.setCpShortTitle(detail.getIrbNumber());
+		srDetail.setType(detail.getSpecimenType());
+		srDetail.setStorageType(SPECIMEN_STORAGE_TYPE_VIRTUAL);
+		srDetail.setAnatomicSite(NOT_SPECIFIED);
+		srDetail.setLaterality(NOT_SPECIFIED);
+		srDetail.setPathology(NOT_SPECIFIED);
+		srDetail.setCollectionProcedure(NOT_SPECIFIED);
+		srDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		srDetail.setName(detail.getProcedureName());
+		srDetail.setInitialQty(new BigDecimal(0));
+		srDetail.setCollectionContainer(detail.getCollectionContainer());
+		srDetail.setSpecimenClass(getSpecimenClass(detail.getSpecimenType()));
+		
+		eventDetail.setSpecimenRequirements(Arrays.asList(srDetail));
+		
+		return eventDetail;
+	}
+
+	private String getSpecimenClass(String specimenType) {
+		ListPvCriteria crit = new ListPvCriteria()
+				.includeParentValue(true)
+				.parentAttribute("specimen_type")
+				.values(Collections.singletonList(specimenType));
+		
+		ResponseEvent<List<PvDetail>> resp = pvSvc.getPermissibleValues(request(crit));
+		List<PvDetail> payload = resp.getPayload();
+		
+		return payload.isEmpty() ? "" : payload.iterator().next().getParentValue();
+	}
+
+	private <T> RequestEvent<T> request(T payload) {
+		return new RequestEvent<T>(payload);
+	}
+	
+	//////////////////////////
+	//
+	// Specimen Requirement
+	//
+	/////////////////////////
+	
+	private void checkSr(CollectionProtocolEventDetail event) throws Exception {
```

Comment:
1. The input to this method should be SpecimenRequirementDetail.
2. Method name should be getSr(SpecimenRequirementDetail input)